packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2,22 +2,1623 @@
+    // less 8.0 dps at the top to account for the vertical space occupied
+    // by the floating library.
+    final double outlineBaseline = aboveBaseline +
+      (containerHeight - (2.0 + aboveBaseline + belowBaseline)) / 2.0;
+
+    double subtextBaseline = 0.0;
+    double subtextHeight = 0.0;
+    if (helperError != null || counter != null) {
+      aboveBaseline = 0.0;
+      belowBaseline = 0.0;
+      layoutLineBox(helperError);
+      layoutLineBox(counter);
+
+      if (aboveBaseline + belowBaseline > 0.0) {
+        const double subtextGap = 8.0;
+        subtextBaseline = containerHeight + subtextGap + aboveBaseline;
+        subtextHeight = subtextGap + aboveBaseline + belowBaseline;
+      }
+    }
+
+    return new _RenderDecorationLayout(
+      boxToBaseline: boxToBaseline,
+      containerHeight: containerHeight,
+      inputBaseline: inputBaseline,
+      outlineBaseline: outlineBaseline,
+      subtextBaseline: subtextBaseline,
+      subtextHeight: subtextHeight,
+    );
+  }
+
+  @override
+  double computeMinIntrinsicWidth(double height) {
+    return _minWidth(icon, height)
+      + contentPadding.left
+      + _minWidth(prefixIcon, height)
+      + _minWidth(prefix, height)
+      + math.max(_minWidth(input, height), _minWidth(hint, height))
+      + _minWidth(suffix, height)
+      + _minWidth(suffixIcon, height)
+      + contentPadding.right;
+  }
+
+  @override
+  double computeMaxIntrinsicWidth(double height) {
+    return _maxWidth(icon, height)
+      + contentPadding.left
+      + _maxWidth(prefixIcon, height)
+      + _maxWidth(prefix, height)
+      + math.max(_maxWidth(input, height), _maxWidth(hint, height))
+      + _maxWidth(suffix, height)
+      + _maxWidth(suffixIcon, height)
+      + contentPadding.right;
+  }
+
+  double _lineHeight(double width, List<RenderBox> boxes) {
+    double height = 0.0;
+    for (RenderBox box in boxes) {
+      if (box == null)
+        continue;
+      height = math.max(_minHeight(box, width), height);
+    }
+    return height;
+    // TODO(hansmuller): this should compute the overall line height for the
+    // boxes when they've been baseline-aligned.
+    // See https://github.com/flutter/flutter/issues/13715
+  }
+
+  @override
+  double computeMinIntrinsicHeight(double width) {
+    double subtextHeight = _lineHeight(width, <RenderBox>[helperError, counter]);
+    if (subtextHeight > 0.0)
+      subtextHeight += 8.0;
+    return contentPadding.top
+      + (label == null ? 0.0 : decoration.floatingLabelHeight)
+      + _lineHeight(width, <RenderBox>[prefix, input, suffix])
+      + subtextHeight
+      + contentPadding.bottom;
+  }
+
+  @override
+  double computeMaxIntrinsicHeight(double width) {
+    return computeMinIntrinsicHeight(width);
+  }
+
+  @override
+  double computeDistanceToActualBaseline(TextBaseline baseline) {
+    // TODO(hansmuller): layout first? Add input.parentData.offset?
```

Comment:
TODO should be actionable

Reply:
This code isn't used so for now it's  `assert(false, 'not implemented')`.